### ISSUE TITLE
A11y: Creators list should be ul instead of dl

### ISSUE THREAD

**Issue by jennur** [inveniosoftware/invenio-requests#362]
![Screenshot 2023-09-15 at 14 07 22](https://github.com/inveniosoftware/invenio-requests/assets/21052053/a68505f5-8beb-433b-9fa6-c9095cd61be8)
![Screenshot 2023-09-15 at 14 07 28](https://github.com/inveniosoftware/invenio-requests/assets/21052053/1812a75c-85a4-4756-ad2b-1bf2fcf3df2b)
